### ISSUE TITLE
`isinstance` check against a tuple that contains a union does not narrow types

### ISSUE THREAD

**Issue by irgolic** [microsoft/pyright#7422]
**Describe the bug**
`isinstance` check against a tuple that contains a union does not narrow types.

**Code or Screenshots**

```python
class A:
    pass


class B:
    pass


class C:
    pass


# correct cases


def main_1(var: A | B | C):
    reveal_type(var)  # Type of "var" is "A | B | C"
    if isinstance(var, A):
        reveal_type(var)  # Type of "var" is "A"
        return
    elif isinstance(var, B | C):
        reveal_type(var)  # Type of "var" is "B | C"
        return
    reveal_type(var)  # Type of "var" is "Never"


def main_2(var: A | B | C):
    reveal_type(var)  # Type of "var" is "A | B | C"
    if isinstance(var, (A, B, C)):
        reveal_type(var)  # Type of "var" is "A | B | C"
        return
    reveal_type(var)  # Type of "var" is "Never"


# incorrect cases


def main_3(var: A | B | C):
    reveal_type(var)  # Type of "var" is "A | B | C"
    if isinstance(var, (A, B | C)):
        reveal_type(var)  # Type of "var" is "A | B | C"
        return
    reveal_type(var)  # Type of "var" is "A | B | C"


def main_4(var: A | B | C):
    reveal_type(var)  # Type of "var" is "A | B | C"
    if isinstance(var, (A | B,)):
        reveal_type(var)  # Type of "var" is "A | B | C"
        return
    reveal_type(var)  # Type of "var" is "A | B | C"

```

This is not particular to `types.UnionType`, it happens with `typing.Union` the same way.

An easy workaround is to check against a union of the tuple (i.e., `isinstance(var, Union[A, B | C])`), but it took me a while to figure out what was going on.

**VS Code extension or command-line**
CLI, v1.1.352

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.353, which I just published.

**Comment by irgolic** [comment]
Thank you very much for the quick fix!
